tomlkit/api.py

```diff
@@ -34,28 +35,28 @@ def loads(string):  # type: (str) -> _TOMLDocument
     return parse(string)
 
 
-def dumps(data, sort_keys=False):  # type: (_TOMLDocument, bool) -> str
+def dumps(data, sort_keys=False):  # type: (Mapping, bool) -> str
     """
     Dumps a TOMLDocument into a string.
     """
-    if not isinstance(data, _TOMLDocument) and isinstance(data, dict):
-        data = item(data, _sort_keys=sort_keys)
+    if not isinstance(data, Container) and isinstance(data, Mapping):
```

Comment:
I changed `TOMLDocument` to `Container` here, because I imagine that structures that are already internal to tomlkit are supposed to rely on `as_string` and that might be the case some of them are already `Mapping`s (maybe TOML tables?).

The changes here are indeed making the function *more acceptive* in terms of arguments, e.g. `MappingProxy` (a Python idiom that makes it easy to pass "immutable dicts" around), `OrderedDict` (although probably the order will not be preserved for Python < 3.6). Alternatively we could also use `Union[dict, TOMLDocument]` to avoid augmenting the function's domain.